Classes/common/touchdb/TD_Database.m

```diff
@@ -76,18 +79,46 @@ static BOOL removeItemIfExists(NSString* path, NSError** outError)
 
 - (NSString*)attachmentStorePath
 {
-    return [[_path stringByDeletingPathExtension] stringByAppendingString:@" attachments"];
+    return [TD_Database attachmentStorePathWithDatabasePath:_path];
 }
 
-+ (TD_Database*)createEmptyDBAtPath:(NSString*)path
++ (NSString *)attachmentStorePathWithDatabasePath:(NSString *)path
+{
+    return [[path stringByDeletingPathExtension] stringByAppendingString:@" attachments"];
+}
+
++ (instancetype)createEmptyDBAtPath:(NSString*)path
+          withEncryptionKeyProvider:(id<CDTEncryptionKeyProvider>)provider
 {
     if (!removeItemIfExists(path, NULL)) return nil;
     TD_Database* db = [[self alloc] initWithPath:path];
     if (!removeItemIfExists(db.attachmentStorePath, NULL)) return nil;
-    if (![db open]) return nil;
+    if (![db openWithEncryptionKeyProvider:provider]) return nil;
     return db;
 }
 
++ (BOOL)isDatabaseEncryptedAtPath:(NSString *)path
+{
+    // Load file
+    NSData *data = [NSData dataWithContentsOfFile:path];
+    if (!data)
+    {
+        return NO;
+    }
+    
+    // Read first 15 bytes
+    char buffer[[TD_DatabaseStandardSQLiteHeader length] + 1];
+    memset(buffer, '\0', sizeof(buffer));
+    
+    [data getBytes:buffer length:(sizeof(buffer) - 1)];
+    
+    // Compare: if the file does not start with the default text, we assume that the
+    // file is encrypted
+    NSString *str = [NSString stringWithCString:buffer encoding:NSASCIIStringEncoding];
+    
+    return ![TD_DatabaseStandardSQLiteHeader isEqualToString:str];
+}
```

Comment:
This method actually checks whether there is a not-unencrypted database at `path` -- it could just as well be any file other than one with a normal SQLite header.

I'd flip the logic of the method and name it `-isDatabaseUnencryptedAtPath:` as that's the only thing you can prove without attempting to decrypt the database.

In addition, could you just read the first `[TD_DatabaseStandardSQLiteHeader length]` bytes from the file rather than the whole thing into an NSData?